tests/components/ViewedByCustomer/ViewedByCustomer.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { ViewedByCustomer } from '../../../src/Index';
+import { Mock, Fake } from 'coveo-search-ui-tests';
+import { IViewedByCustomerOptions } from '../../../src/components/ViewedByCustomer/ViewedByCustomer';
+import { IQueryResult } from 'coveo-search-ui';
+import { AdvancedComponentSetupOptions } from 'coveo-search-ui-tests/MockEnvironment';
+
+describe('ViewedByCustomer', () => {
+    const VIEWED_BY_CUSTOMER_LABEL_CSS_CLASS = 'viewed-by-customer-label';
+    const VIEWED_BY_CUSTOMER_ICON_CSS_CLASS = 'viewed-by-customer-icon';
+
+    describe('when the field isUserActionView of the result true', () => {
+        const fakeResult: IQueryResult = { ...Fake.createFakeResult(), isUserActionView: true };
+
+        it('should display the default label and the icon when no options are given', () => {
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult).cmp;
+            expect(testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_ICON_CSS_CLASS).length).toBe(1);
+            expect(testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_LABEL_CSS_CLASS).length).toBe(1);
+            expect((testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_LABEL_CSS_CLASS).item(0) as HTMLSpanElement).innerText).toBe(
+                'Viewed by Customer'
+            );
+        });
+
+        it('should display the label given in the option if any is given', () => {
+            const option = new Mock.AdvancedComponentSetupOptions();
+            (option.cmpOptions as IViewedByCustomerOptions).label = 'test label';
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult, option).cmp;
+            expect(testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_LABEL_CSS_CLASS).length).toBe(1);
+            expect(testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_LABEL_CSS_CLASS).item(0).innerHTML).toBe('test label');
+        });
+
+        it('should hide the icon if showIcon option is set to false', () => {
+            const option = new Mock.AdvancedComponentSetupOptions();
+            (option.cmpOptions as IViewedByCustomerOptions).showIcon = false;
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult, option).cmp;
+            expect(testComponent.element.getElementsByClassName(VIEWED_BY_CUSTOMER_ICON_CSS_CLASS).length).toBe(0);
+        });
+    });
+
+    [false, undefined, null].forEach(falsyValue => {
+        describe(`when the field isUserActionView of the result is ${falsyValue}`, () => {
+            const fakeResult: IQueryResult = { ...Fake.createFakeResult(), isUserActionView: falsyValue };
+
+            it('nothing should should be displayed or added to the DOM', () => {
```

Comment:
Rephrase the test to start with `it should...` rather than `it nothing...` something like `it should not display the component`